### PR TITLE
test: add e2e test for region failover

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
#4161 
wait for #4172
## What's changed and what's your intention?

- Modified `CreateTableExprGenerator` to use `Ident` instead of `String` for table names in `create_expr.rs`
- Updated `InsertExprGenerator` to use `Ident` instead of `String` for table names in `insert_expr.rs`
- Introduced new utility modules: `cluster_info`, `partition`, and `wait` in `utils.rs`
- Added `GT_FUZZ_CLUSTER_NAMESPACE` and `GT_FUZZ_CLUSTER_NAME` constants in `utils.rs`
- Created new fuzz target `fuzz_failover_mito_regions.rs`
- Implemented `Ident::is_empty` method in `ir.rs`
- Refactored timestamp generation for MySQL to use `TsValueGenerator`
- Added `count_values` function in `validator/row.rs`
- Updated `Makefile` to include `fuzz-ls` target for listing fuzz targets

```mermaid
sequenceDiagram
    participant Client
    participant Cluster
    participant Datanode
    participant K8s

    Client->>Cluster: Initialize connections
    Cluster->>Datanode: Fetch node info
    Datanode-->>Cluster: Return node info
    Cluster->>Client: Return connections
    Client->>K8s: Simulate pod failure
    Cluster-->>Datanode: Region failover
    Client->>K8s: Recover pod
    Client->>Cluster: Validate data consistency
    Cluster-->>Client: Data validation result
```

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Introduced functionality for fetching cluster node information, managing partitions, and simulating pod failures.
  - Added a fuzzing target for testing failover scenarios, which includes generating table schemas, inserting data, and validating data consistency post-recovery.

- **Refactor**
  - Updated methods for MySQL connection options and logging configurations.

- **Enhancements**
  - Added utility functions for waiting for conditions, managing regions, and dealing with pod failures in a Kubernetes cluster.

### CODE REVIEW

File: tests-fuzz/src/utils/wait.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        loop {
+            if condition(check().await) {
+                break;
+            }
+            tokio::time::sleep(retry_interval).await
+        }
+    })
+    .await
+    .unwrap();
+}
```
[the remark this answers is not in the document]
Comment (WenyXu) [reply tests-fuzz/src/utils/wait.rs:38]: It's ok, just for the test.

File: .github/actions/setup-greptimedb-cluster/action.yml [hunk 2]
```diff
@@ -24,7 +24,8 @@ inputs:
     description: "Etcd endpoints"
   values-filename:
     default: "with-minio.yaml"
-    
+  enable-region-failure:
+    default: false
```
[the remark this answers is not in the document]
Comment (WenyXu) [reply .github/actions/setup-greptimedb-cluster/action.yml:28]: @coderabbitai generate the description

**Comment by WenyXu** [comment]
@coderabbitai resolve

File: tests-fuzz/targets/failover/fuzz_failover_mito_regions.rs [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+        ctx.kube.clone(),
+        &ctx.namespace,
+        &ctx.cluster_name,
+        selected_datanode,
+        360,
+    )
+    .await?;
+
+    // Waits for num of regions on `selected_datanode` become to 0.
+    wait_for_all_regions_evicted(
```
Comment (zhongzc) [inline tests-fuzz/targets/failover/fuzz_failover_mito_regions.rs:306]: Will the data be unavailable during this period?

**Review by zhongzc — APPROVED** [review]

**Comment by WenyXu** [comment]
https://github.com/coderabbitai resolve

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**